File: app/src/main/java/fr/android/androidexercises/booklist/BookItemView.java
package fr.android.androidexercises.booklist;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import fr.android.androidexercises.R;
import fr.android.androidexercises.model.Book;

public class BookItemView extends LinearLayout {
    private TextView name;
    private TextView price;

    public BookItemView(Context context) {
        this(context, null);
    }

    public BookItemView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public BookItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        name = findViewById(R.id.nameTextView);
        price = findViewById(R.id.priceTextView);
    }

    public View bind(Book book) {
        name.setText(book.getTitle());
        price.setText(String.valueOf(book.getPrice()));

        return this;
    }
}

File: app/src/main/java/fr/android/androidexercises/BookItemView.kt
package fr.android.androidexercises

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import kotlinx.android.synthetic.main.custom_view_item_book.view.*


/**
 * Created by damien on 10/11/2017.
 */
class BookItemView @JvmOverloads constructor(context: Context,
                                             attrs: AttributeSet? = null,
                                             defStyleAttr: Int = 0) :
        LinearLayout(context, attrs, defStyleAttr), ItemView<Book> {

    override fun bind(element: Book): View {
        nameTextView.text = element.name
        priceTextView.text = element.price.toString()
        return this
    }
}